CHANGELOG.md

```diff
@@ -12,6 +12,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Simulation user guide
 - Example for transfer learning backtest utility
 
+### Removed
+- Optional `telemetry` dependency is no longer a group
```

Comment:
if what @AdrianSosic wrote here (https://github.com/emdgroup/baybe/issues/179#issuecomment-2017836903) is true this entire fix would only be half-way. Poetry would prob still fail if other optional groups are used (as quite some of them have references to other groups).

One way would be to remove self-referencing groups altogether which however would drastically reduce the overview and maintainalbility inthe toml file

Another way would be to recommend that people do not use terrible package managers

Putting this here for discussion

Reply:
Don't let certain people read the comment regarding "terrible package managers" :D
I tested installing other optional dependencies, and at least during my tests, *all of that worked*. The only things that seemed to be problematic was "having an optional dependency in the main dependencies". However, this should be verified in any case by somebody who is actually more familiar with poetry.
I'd however also be fine to explicitly discourage people from using `poetry`. In that case, we could re-purpose this PR and just add a few sentences about this in `known_issues` and the installation part of `README`

Reply:
oh well in that case the problem appears fixed

imo no need to add to known issues, thats for permanent issues we cannot fix or circumvent

Reply:
Installing with poetry works for me, both the version with no extra deps

`poetry add git+https://github.com/emdgroup/baybe@c9fb0c5c17dc51cc8d630c877eebe58d9b10f37d`

as well as the version with extra deps where the dependency group pulls in other dependency groups

`poetry add git+https://github.com/emdgroup/baybe@c9fb0c5c17dc51cc8d630c877eebe58d9b10f37d[examples]`

@Scienfitz: Maybe just do not use *optional* dependencies to declare *required* dependencies 😅

Reply:
Ok, so then this should be safe to merge once @AdrianSosic has approved.

Reply:
great thanks for confirming @tobiasploetz 
this way of declaration was done solely to keep a better overview (in particular telemetry unfortunately requires several packages), the group function is very useful for that (and most reasonable package managers have no issues with that ;) )

Reply:
Nice thanks to all of you.
@Scienfitz: sorry for my misguiding comment at the beginning: when I typed it, I was not aware that we have that one `telemetry` group in the main deps